src/value.c

```diff
@@ -192,7 +251,66 @@ size_t SIValue_StringConcat(SIValue *strings, unsigned int string_count, char *b
 	return offset;
 }
 
+// assumption: either a or b is a string
+SIValue SIValue_ConcatString(const SIValue a, const SIValue b) {
+	SIValue result;
+	char buffer[512];
+	char *string_arg = NULL;
+	// a is not a string - scalar + string
+	if(a.type != T_STRING) {
+		/* a is numeric, convert to string. */
+		SIValue_ToString(a, buffer, 512);
+		result = SI_DuplicateStringVal(buffer);
+	}
+	// string + value
+	else result = SI_Clone(a);
+
+	unsigned int argument_len = 0;
+	if(b.type != T_STRING) {
+		/* b is not a numeric, get a string representation. */
+		argument_len = SIValue_ToString(b, buffer, 512);
+		string_arg = buffer;
+	} else {
+		string_arg = b.stringval;
+		argument_len = strlen(string_arg);
```

Comment:
if I'll do it there will be a redundant call to `strlen` in case `b` is not a string. The returned value from `SIValue_ToString` is the length of the string